### PR TITLE
Anya/2090 poa name from rep table

### PR DESCRIPTION
**Pull request by aroltsch**
Connects #2090

### CODE REVIEW

File: app/mappers/power_of_attorney_mapper.rb [hunk 1]
```diff
@@ -38,11 +43,9 @@ def get_poa_from_vacols_poa(vacols_code)
     else
       # Otherwise we have to look up the specific name of the rep
       # in the REP table.
-      # TODO: modify poa repository to look it up and pass that
-      # info into this method.
       {
-        representative_name: "Stub POA Name",
-        representative_type: "Stub POA Type"
+        representative_name: get_rep_name_from_rep_record(representative_record),
+        representative_type: get_short_name(vacols_code)
```
Comment (amprokop) [inline app/mappers/power_of_attorney_mapper.rb]: looks great

Comment (amprokop) [inline app/mappers/power_of_attorney_mapper.rb]: was a little confused by this at first, but we'll only reach this part in the conditional if vacols_code corresponds to   Attorney, Agent, Other, One Time Rep, so I think this is good!

File: app/services/power_of_attorney_repository.rb [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
   # :nocov:
 
   # returns either the data or false
-  #
   def self.load_vacols_data(poa)
     case_record = MetricsService.record("VACOLS POA: load_vacols_data #{poa.vacols_id}",
                                         service: :vacols,
                                         name: "PowerOfAttorneyRepository.load_vacols_data") do
-      poa_query
+      poa_query(poa)
```
Comment (amprokop) [inline app/services/power_of_attorney_repository.rb]: word

File: lib/fakes/power_of_attorney_repository.rb [hunk 3]
```diff
@@ -5,9 +5,13 @@ class FakePoaRecord
     def self.bfso
       "A"
     end
+
+    def self.representative
+      nil
```
Comment (amprokop) [inline lib/fakes/power_of_attorney_repository.rb]: do we want the test data to be `nil`?

Comment (amprokop) [inline lib/fakes/power_of_attorney_repository.rb]: fine if we do

Comment (aroltsch) [inline lib/fakes/power_of_attorney_repository.rb]: In this case I think it is okay. Because representative can be nil for a case_record.

**Pull request merged**